charon/cmd/cmd_merge.py

```diff
@@ -0,0 +1,154 @@
+"""
+Copyright (C) 2022 Red Hat, Inc. (https://github.com/Commonjava/charon)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import List
+
+from charon.utils.archive import detect_npm_archives, NpmArchiveType
+from charon.cmd.internal import _get_local_repos, _decide_mode
+from charon.pkgs.maven import _extract_tarballs
+from click import command, option, argument
+from zipfile import ZipFile, ZIP_DEFLATED
+from tempfile import mkdtemp
+
+import logging
+import os
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+@argument(
+    "repos",
+    type=str,
+    nargs=-1  # This allows multiple arguments for zip urls
+)
+@option(
+    "--product",
+    "-p",
+    help="""
+    The product key, will combine with version to decide
+    the metadata of the files in tarball.
+    """,
+    nargs=1,
+    required=True,
+    multiple=False,
+)
+@option(
+    "--version",
+    "-v",
+    help="""
+    The product version, will combine with key to decide
+    the metadata of the files in tarball.
+    """,
+    required=True,
+    multiple=False,
+)
+@option(
+    "--root_path",
+    "-r",
+    default="maven-repository",
+    help="""
+    The root path in the tarball before the real maven paths,
+    will be trailing off before uploading.
+    """,
+)
+@option(
+    "--work_dir",
+    "-w",
+    help="""
+    The temporary working directory into which archives should
+    be extracted, when needed.
+    """,
+)
+@option(
+    "--merge_result",
+    "-m",
+    help="""
+    The path of the final merged zip file will be compressed and saved.
```

Comment:
maybe we should give a note that if this is not provided which default result will be used?